test/unit/SingletonConfiguration.php

```diff
@@ -0,0 +1,27 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\BetterReflectionTest;
+
+use Roave\BetterReflection\Configuration;
+
+/**
+ * This is a static class needed to share configuration among tests.
+ *
+ * The reason for doing so is to just speed them up internally, nothing else.
+ *
+ * All tests should run the same whether the `configuration()` method returns
+ * a new instance of a shared one.
+ */
+final class SingletonConfiguration
+{
+    /**
+     * @var Configuration|null
+     */
+    private static $configuration;
+
+    public static function configuration() : Configuration
```

Comment:
This makes the test suite ~40% faster, so I think it's worth keeping it for now.